### PR TITLE
Feature/add devise

### PR DESCRIPTION
**Pull request by Metaverse-Nabeel**
### The following changes have been made to this PR:
 
- [x] Devise setup / Install
- [x]  Sign up a new user
- [x]  Post as a user
- [x]  Fix linter errors
- [x] Update Readme

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    User.first
-  end
+  # def current_user
+  #   User.first
+  # end
```
Comment (cynthiainga) [inline app/controllers/application_controller.rb]: - [x] You should remove this method and implement a new method provided by `Devise` to configure Devise, please kindly add it and make sure the `Devise` is setup correctly using the [getting started link](https://github.com/heartcombo/devise) 🙏🏾 
- [x] The requirements say users should register on the app using email and password, you can add other things like "name, bio, image URL" unfortunately, I was able to access the app pages without login or signup. Kindly fix this issue.
- [x] Also make sure that the app work correctly, after user login or signup he can create a post, add a comment, and like a post 🙃 😉

File: app/models/user.rb [hunk 2]
```diff
@@ -1,7 +1,20 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  before_validation :set_default
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
```
Comment (cynthiainga) [inline app/models/user.rb]: - [x] The devise is written twice please kindly fix that 🙏🏾

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @Metaverse-Nabeel & @Booze33, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Good commit history ✔️ 
- [ ] Correct Gitflow ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 2 above)
Comment (Metaverse-Nabeel) [reply app/models/user.rb]: Done

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (Metaverse-Nabeel) [reply app/controllers/application_controller.rb]: All the task are done as per your review.

File: app/models/user.rb (hunk 2 above)
Comment (Metaverse-Nabeel) [reply app/models/user.rb]: Done

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @Metaverse-Nabeel , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed
- [ ] devise installed succesfully
- [ ] app works fine however, I can no longer create a post

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**